homeassistant/components/yolink/binary_sensor.py

```diff
@@ -58,70 +59,122 @@ class YoLinkBinarySensorEntityDescription(BinarySensorEntityDescription):
 ]
 
 
+def parse_data_leak_sensor_state(device: YoLinkDevice, data: dict) -> bool | None:
+    """Parse leak sensor state."""
+    if device.device_type == ATTR_DEVICE_MULTI_CAPS_LEAK_SENSOR:
+        if (state := data.get("state")) is None or (
+            value := state.get("waterDetection")
+        ) is None:
+            return None
+        return (
+            value == "normal"
+            if data.get("waterDetectionMode") == "WaterPeak"
+            else value == "leak"
+        )
+    return data.get("state") in ["alert", "full"]
+
+
+def is_leak_sensor_state_available(device: YoLinkDevice, data: dict) -> bool:
+    """Check leak sensor state availability."""
+    if device.device_type == ATTR_DEVICE_MULTI_CAPS_LEAK_SENSOR:
+        if (
+            (alarms := data.get("alarm")) is not None
+            and isinstance(alarms, list)
+            and "Alert.DetectorError" in alarms
+        ):
+            return False
+    if (alarms := data.get("alarmState")) is not None and alarms.get(
+        "detectorError"
+    ) is True:
+        return False
+    return True
```

Comment:
Would it make sense to maybe split out the entity description into 2? As in, now we try to reuse the same entity description, but we have to mingle all this logic into it for it to work

Reply:
For different models of device of the same type, using the same entity description seems more reasonable. The data differences between models are distinguished within the `parse_data_leak_sensor_state` and `is_leak_sensor_state_available` functions.